eclipse-theia/theia
# plugin: Support LogOutputChannel

#### What it does


- Support LogOutputChannel
- Support LogLevel
- Support in namespace/env: logLevel & onDidChangeLogLevel

Resolves #12017

Contributed on behalf of STMicroelectronics.

##### Remark:
This PR contributes the basic VS Code API support for LogOutputChannels, to fully support all aspects of a LogOutputViewChannel as in VS Code some additional features will be needed:
  - In VSCode there is support to set the "Developer log level" in the application which also configures the log level for LogOutputviewChannels. Currently the log level for LogOutputChannels is set to the default log level, i.e. info.
  - VSCode also writes the logs to a separate file per extension and offers a command and tool in the output view to directly open the extensions's log file.
 
I suggest to extract those additional features into separate follow-up issues if you agree.

[Edit 2023-04-21]: Extracted to follow-up issue #12440

#### How to test


To test, I created a simple extension that creates a LogOutputChannel and contributes commands to send log messages in different levels:
[Edit 2023-04-21: updated VSIX test extension] [vscode-extension-log-output-channel-12017-0.0.1.zip](https://github.com/eclipse-theia/theia/files/11294198/vscode-extension-log-output-channel-12017-0.0.1.zip)



1. Download and install the test extension `vscode-extension-log-output-channel-12017-0.0.1.vsix` in your Theia test workspace (via `Extensions` view - top menu bar `More actions...` - `Install from VSIX`).
2. On activation of the extension, the log output channel `LogOutputChannel-Test-12017` is shown in the output view and the initial message (`LogLevel set to: Info`) should be shown.
3. Execute the commands (`Log a <loglevel> message via LogOutputChannel (with args)` / `Throw an ERROR and log msg/stack via LogOutputChannel` / `Test replace LogOutputChannel` / `Append text content (as new line) to LogOutputChannel`) via the command palette and check for the messages in the output channel.
Remark: Currently the loglevel is set to default INFO log level.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
